sql/core/src/test/scala/org/apache/spark/sql/execution/command/DropNamespaceSuiteBase.scala

```diff
@@ -0,0 +1,114 @@
+
+/**
+ * This base suite contains unified tests for the `DROP NAMESPACE` command that check V1 and V2
+ * table catalogs. The tests that cannot run for all supported catalogs are located in more
+ * specific test suites:
+ *
+ *   - V2 table catalog tests: `org.apache.spark.sql.execution.command.v2.DropNamespaceSuite`
+ *   - V1 table catalog tests: `org.apache.spark.sql.execution.command.v1.DropNamespaceSuiteBase`
+ *     - V1 In-Memory catalog: `org.apache.spark.sql.execution.command.v1.DropNamespaceSuite`
+ *     - V1 Hive External catalog: `org.apache.spark.sql.hive.execution.command.DropNamespaceSuite`
+ */
+trait DropNamespaceSuiteBase extends QueryTest with DDLCommandTestUtils {
+  override val command = "DROP NAMESPACE"
+
+  protected def builtinTopNamespaces: Seq[String] = Seq.empty
+  protected def isCasePreserving: Boolean = true
+  protected def assertDropFails
+
+  protected def checkNamespace(expected: Seq[String]) = {
+    val df = spark.sql(s"SHOW NAMESPACES IN $catalog")
+    assert(df.schema === new StructType().add("namespace", StringType, false))
+    checkAnswer(df, expected.map(Row(_)))
+  }
+
+  test("basic tests") {
+    sql(s"CREATE NAMESPACE $catalog.ns")
+    checkNamespace(Seq("ns") ++ builtinTopNamespaces)
+
+    sql(s"DROP NAMESPACE $catalog.ns")
+    checkNamespace(builtinTopNamespaces)
+  }
+
+  test("DropNamespace: test handling of 'IF EXISTS'") {
+    // It must not throw any exceptions
+    sql(s"DROP NAMESPACE IF EXISTS $catalog.unknown")
+    checkNamespace(builtinTopNamespaces)
+  }
+
+  test("DropNamespace: Namespace does not exist") {
+    // Namespace $catalog.unknown does not exist.
+    val message = intercept[AnalysisException] {
+      sql(s"DROP DATABASE $catalog.unknown")
+    }.getMessage
+    assert(message.contains(s"'unknown' not found"))
+  }
+
+  test("DropNamespace: drop non-empty namespace with a non-cascading mode") {
+    sql(s"CREATE NAMESPACE $catalog.ns")
+    sql(s"CREATE TABLE $catalog.ns.table (id bigint) $defaultUsing")
+    checkNamespace(Seq("ns") ++ builtinTopNamespaces)
+
+    // $catalog.ns.table is present, thus $catalog.ns cannot be dropped.
+    assertDropFails
+    sql(s"DROP TABLE $catalog.ns.table")
+
+    // Now that $catalog.ns is empty, it can be dropped.
+    sql(s"DROP NAMESPACE $catalog.ns")
+    checkNamespace(builtinTopNamespaces)
+  }
+
+  test("DropNamespace: drop non-empty namespace with a cascade mode") {
+    sql(s"CREATE NAMESPACE $catalog.ns")
+    sql(s"CREATE TABLE $catalog.ns.table (id bigint) $defaultUsing")
+    checkNamespace(Seq("ns") ++ builtinTopNamespaces)
+
+    sql(s"DROP NAMESPACE $catalog.ns CASCADE")
+    checkNamespace(builtinTopNamespaces)
+  }
+
+  test("DropNamespace: drop current namespace") {
+    sql(s"CREATE NAMESPACE $catalog.ns")
+    sql(s"USE $catalog.ns")
+    sql(s"DROP NAMESPACE $catalog.ns")
+    checkNamespace(builtinTopNamespaces)
+  }
+
+  test("DropNamespace: drop namespace with case sensitivity") {
```

Comment:
nit: this is the drop namespace test suite, do we really need to put the `DropNamespace: ` prefix in all the test names here?

Reply:
yeah, will remove it. Thanks!